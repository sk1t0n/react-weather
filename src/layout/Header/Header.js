import Navbar from 'react-bootstrap/Navbar';
import { Container, Nav } from 'react-bootstrap';
import { NavLink } from 'react-router-dom';

const Header = (props) => {
  const links = [
    { title: 'Home', to: '/', exact: true },
    { title: 'Info', to: '/info', exact: false }
  ];

  return (
    <header>
      <Navbar bg="dark" variant="dark" expand="sm">
        <Container>
          <Navbar.Brand href="/">React-Weather</Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="me-auto">
              {links.map(link => (
                <NavLink
                  key={link.title}
                  className={'nav-link'}
                  to={link.to}
                  exact={link.exact}
                >
                  { link.title }
                </NavLink>
              ))}
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    </header>
  );
}

export { Header };
