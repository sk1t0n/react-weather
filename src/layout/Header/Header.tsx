import React from 'react';
import Navbar from 'react-bootstrap/Navbar';
import { Container, Nav } from 'react-bootstrap';
import { NavLink } from 'react-router-dom';

export const Header: React.FC = () => {
  const links = [
    { title: 'Home', to: '/', exact: true },
    { title: 'Info', to: '/info', exact: false }
  ];

  return (
    <header>
      <Navbar bg="dark" variant="dark" expand="sm">
        <Container>
          <Navbar.Brand href="/">React-Weather</Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="me-auto">
              {links.map(link => (
                <NavLink
                  key={link.title}
                  className={'nav-link'}
                  to={link.to}
                  exact={link.exact}
                >
                  { link.title }
                </NavLink>
              ))}
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    </header>
  );
}
