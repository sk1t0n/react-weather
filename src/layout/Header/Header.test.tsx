import { render, screen } from '@testing-library/react';
import { Header } from './Header';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import userEvent from '@testing-library/user-event';

describe('Header component', () => {
  it('check the snapshot', () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router history={history}>
        <Header />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  describe('Testing router', () => {
    it('check exist the class with name "active"', () => {
      const history = createMemoryHistory();
      history.push('/');
      render(
        <Router history={history}>
          <Header />
        </Router>
      );

      const linkHome = screen.getByText(/home/i);
      expect(linkHome).toHaveClass('active');
      userEvent.click(screen.getByText(/info/i));
      expect(linkHome).not.toHaveClass('active');
    });
  });
});
