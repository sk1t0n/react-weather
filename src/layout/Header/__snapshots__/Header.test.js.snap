// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component check the snapshot 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          React-Weather
        </a>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
            >
              Home
            </a>
            <a
              class="nav-link"
              href="/info"
            >
              Info
            </a>
          </div>
        </div>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
